kore/src/Kore/Rewrite/Search.hs

```diff
@@ -153,9 +153,8 @@ matchWith sideCondition e1 e2 = do
                     ]
                     [Conditional.substitution predSubst]
             let simplified = merged
-            smtEvaluation <-
-                lift $ SMT.Evaluator.evaluate simplified
-            case smtEvaluation of
+            evaluated <- lift $ evalConditional simplified
```

Comment:
Are we sure this doesn't need `sideCondition`? In other words, `evalConditionalWithSideCondition simplified sideCondition`.

Reply:
I will add the side condition and post failing unit test.

Reply:
<details>
<summary>Unit test failures with side condition</summary>
<p>

```
↓ test/Driver.hs                                                                    
↓ Test                                                                              
↓ Kore                                                                              
↓ Exec                                                                              
↓ searchPriority                                                                    
✗ searchPriority ONE                                                                
                                                                                    
Test threw an exception                                                             
Failed to decide predicate:                                                         
    /* Spa */                                                                       
    \equals{MySort{}, _}(                                                           
        /* Fl Fn D Sfa */ V:MySort{},                                               
        /* Fl Fn D Sfa Cl */ c{}()                                                  
    )                                                                               
with side condition:                                                                
    /* D Sfa */ \top{_}()

↓ test/Driver.hs                                                                    
↓ Test
↓ Kore
↓ Exec
↓ searchPriority
✗ searchPriority STAR

Test threw an exception
Failed to decide predicate:
    /* Spa */
    \equals{MySort{}, _}(
        /* Fl Fn D Sfa */ V:MySort{},
        /* Fl Fn D Sfa Cl */ a{}()
    )
with side condition:
    /* D Sfa */ \top{_}()

↓ test/Driver.hs                                                          [132/1980]
↓ Test
↓ Kore
↓ Exec
↓ searchPriority
✗ searchPriority PLUS

Test threw an exception
Failed to decide predicate:
    /* Spa */
    \equals{MySort{}, _}(
        /* Fl Fn D Sfa */ V:MySort{},
        /* Fl Fn D Sfa Cl */ c{}()
    )
with side condition:
    /* D Sfa */ \top{_}()

↓ test/Driver.hs                                                                    
↓ Test                                                                              
↓ Kore
↓ Exec
↓ searchPriority
✗ searchPriority FINAL

Test threw an exception
Failed to decide predicate:
    /* Spa */
    \equals{MySort{}, _}(
        /* Fl Fn D Sfa */ V:MySort{},
        /* Fl Fn D Sfa Cl */ d{}()
    )
with side condition:
    /* D Sfa */ \top{_}()
```

</p>
</details>